src/interactive-window/editor-integration/codeLensFactory.ts

```diff
@@ -138,13 +138,15 @@ export class CodeLensFactory implements ICodeLensFactory {
                 commands.forEach((c) => {
                     const codeLens = this.createCodeLens(document, r, c, firstCell);
                     if (codeLens) {
-                        cache?.documentLenses.push(codeLens); // NOSONAR
+                        cache!.documentLenses.push(codeLens); // NOSONAR
```

Comment:
This seems like such an odd typescript limitation. I poked around a bit, but didn't turn up a good quick explanation. Anyone happen to know the quick answer for needing the "!" here?

Reply:
Code is fine BTW, just wondering about the language.

Reply:
I'm guessing the forEach's scope disrupts the type narrowing done earlier. I can't see a reason why it would, so it's just probably hard to implement